### PR TITLE
Auto-Scrolling Chat

### PR DESCRIPTION
**Pull request by gmelsby**
Automatically scrolls to the bottom of the chat whenever the div containing the llama response expands. The user does not have to manually scroll to read the response from the chat. 

In order to implement this, the user input text box is fixed to the bottom of the screen.

### CODE REVIEW

File: chat.js [hunk 1]
```diff
@@ -96,6 +103,8 @@ async function submitRequest() {
           }
           responseDiv.hidden_text += word;
           responseDiv.innerHTML = DOMPurify.sanitize(marked.parse(responseDiv.hidden_text)); // Append word to response container
+          // autoscrolls chat container
+          scrollToBottom(chatContainer);
```
Comment (rtcfirefly) [inline chat.js]: Do we need this extra call to scroll to the bottom?
If the user has already scrolled up then they are probably looking at previous chats

**Comment by rtcfirefly** [comment]
Thanks for the PR!

When I tested it, it seems like the autoscroll takes over even when the user scrolls away to check old conversations. Would you be able to fix this?

Also, it would be good to rebase on main and squash the commits for a cleaner history.

**Comment by gmelsby** [comment]
Thanks for checking this out! I'll make some fixes and rebase + squash. 

I think I can implement toggling the autoscroll with some scroll event listeners--i.e. if the user scrolls up, autoscroll turns off.

I'm also pretty sure the floating message box effect can be fixed by adding a background to the div containing it. I'll check!

I also made a little quality-of-life change last night that pushes the name of the selected model into the query parameters of the page's URL. 
If a user selects orca:latest for example, the URL becomes `http://localhost:8000?model=orca%3Alatest` and orca:latest is auto-selected in the model select drop down upon refresh.
If this is a feature you'd like let me know and I can include it in the PR with the autoscroll fixes!

**Comment by rtcfirefly** [comment]
Yep, that sounds like a nice feature to add 👍

**Comment by gmelsby** [comment]
Made some adjustments and rebased+squashed on the main branch of my fork. I'll create a new PR

**Pull request closed without merging**